api/src/main/java/com/google/appengine/api/datastore/DatastoreApiHelper.java

```diff
@@ -130,23 +130,25 @@ static <T extends Message> Future<T> makeAsyncCall(
     return new FutureWrapper<byte[], T>(response) {
       @Override
       protected T wrap(byte[] responseBytes) throws InvalidProtocolBufferException {
+        T updatedResponseProto = null;
         // This null check is mainly for the benefit of unit tests
         // (specifically ones using EasyMock, where the default behavior
         // is to return null).
         if (responseBytes != null && responseProto != null) {
+
           try {
-            responseProto.getParserForType().parseFrom(responseBytes);
+            updatedResponseProto = (T) responseProto.getParserForType().parseFrom(responseBytes);
```

Comment:
Can you explain why we need a new variable?

Reply:
I am trying to pass the response argument as a builder in this func, then we would not need another variable. But facing a bit of difficulty in handling the other functions which calls this makeAsyncCall method